### DISCUSSION TITLE
Drag event across pages?

### DISCUSSION THREAD

**Discussion by Bertie2011** [vkurko/calendar#334] [Q&A]
Would it be possible to allow events to be dragged across different "pages" of a view?

For example, we're currently using the `resourceTimelineMonth` view and although by default the calendar view scrolls as you drag the event, it does not automatically switch to other months.

I attempted to hack my way around this with a mouse-leave event, checking some coordinates and calling `calendar.next()` and `calendar.prev`. It does show the ghost of the event on the next month view, but once dropped it does not appear in the view and the previous month page (where the event originated from) wasn't updated either.

**Comment by vkurko** [comment]
Thanks, this idea seems interesting. If I have some free time or if I get more requests to implement this feature, I will try to implement something.

**Reply by Bertie2011** [reply]
Is there any quick work around/solution you might have in the meantime?

**Reply by vkurko** [reply]
Maybe implement a cut/paste functionality? Some button inside the event or in the dropdown menu to "cut" this event. And then anywhere else in the calendar you can "paste" this event using `dateClick`.

**Reply by digidoed** [reply]
when I read this post and it did seem interesting, I tried to do it on my implementation of this calendar and it worked, but not totally... In my version I have built in keyboardshortcuts for the prev and next buttons. So when I start dragging the event and switch the view with the keyboard shortcut and then drop it. it has been moved, BUT(!!) it doesn't show directly, only after a reload of the events in the view. This because of the fact that I do an update of the event with the available updateEvent(e). I guess if you check when onEventDragStopped, whether the view is the same as, the view you would have to set on onEventDragStarted and if not the same do a reload of the events in the dropped view, it would work. So that could be a workaround, even though it is over a year ago, you needed this.
For MegaMaster vkurko a thought for when implementing this as a feature hovering at or beyond the start or end of the view would cause prev or next to trigger, or for anyone that wants to implement this as a feature. 
I myself use the standalone version of this wonderful set of components as a nuxt plugin, so I use vue and stuff as it is more my domain and the domain this is where it had been implemented.

Hope this helps others to achieve to implement this functionality.

**Reply by digidoed** [reply]
Sorry, when I said to check for the view that changed, I actually meant to check for a change in e.g. the views currentStart or something to know that the period/ range of the view has changed (dragging from one view to another type of view (might be practical in some cases as well though).
I can confirm it works as I just added this change and therefore can confirm it does work for the event that is being dragged, except for one little issue. When the view changed to the prev or next while the event is being dragged, this 'ghost' event is not being rendered only after the mouse is being moved. I understand kind of why this happens, but do not know exactly where to solve this, as it would have to be done in the svelte part and I use the standalone version...
For the full draggable implementation I would like to ask our Hero vkurko for a suggestion (or feature/ enhancement request) for detecting hovering over the start or end of a view, while dragging.

**Reply by vkurko** [reply]
@digidoed Thanks for the suggestions. It seems like a feature like this hasn't been implemented in FullCalendar for [10 years](https://github.com/fullcalendar/fullcalendar/issues/2542). I'll try to implement it a little faster :smile:

**Comment by creationcz** [comment]
Something new in this topic? Functionality would be very useful to me. Or is it in plan? If not, i will use cut/paste functionality mentioned above

**Reply by vkurko** [reply]
> Something new in this topic? Functionality would be very useful to me. Or is it in plan? If not, i will use cut/paste functionality mentioned above

Just finished the time consuming task of migrating to Svelte 5. Now I'll try to implement the accumulated small tasks, but, frankly, the above task is not a priority. I advise to go the cut/paste way for now.
